### PR TITLE
[wip] Set eliminate=False per default in hyper()

### PR DESCRIPTION
**Pull request by skirpichev**
Not sure how common cancellation is to affect performance badly.  Hence, here is a simplest solution for #349.

Except one test, nothing seems to be affected in your test suite.  But please note, that moderate amount of docstrings - not tested.

Also readme update about Code of Conduct.

### CODE REVIEW

**Comment by fredrik-johansson** [comment]
Could you please revert the addition to the README? I don't think it needs to be said that contributions will be judged by their technical merit, and a link to someone's opinion piece on codes of conduct (even if I largely agree with the sentiment behind it) is plainly off topic.

**Comment by fredrik-johansson** [comment]
The performance issue matters, but I can easily fix it later if you don't want to bother with it.

File: mpmath/function_docs.py [hunk 1]
```diff
@@ -2816,6 +2816,15 @@
     >>> hyper([1,1,1,-1],[-2,5],1)
     1.1
 
+With the keyword argument ``eliminate=True``, function degree reduced
```
Comment (fredrik-johansson) [inline mpmath/function_docs.py]: "the function degree is"

File: mpmath/function_docs.py [hunk 2]
```diff
@@ -2816,6 +2816,15 @@
...
+first by cancellation of common parameters.  For nonpositive parameters
+that may lead to the different function definition::
```
Comment (fredrik-johansson) [inline mpmath/function_docs.py]: "to a different"

**Comment by skirpichev** [comment]
> I don't think it needs to be said that contributions will be judged by their technical merit

So, you wont formalize your words (quoted in the commit message) in project docs, in any from?

**Comment by fredrik-johansson** [comment]
> So, you wont formalize your words (quoted in the commit message) in project docs, in any from?

I don't think it belongs.

**Comment by skirpichev** [comment]
Ok, sorry.  Lets wait if someone else will come with better solution.

**Pull request closed without merging**